flexmeasures/utils/coding_utils.py

```diff
@@ -1,5 +1,8 @@
 import functools
 import time
+import inspect
+import warnings
```

Comment:
I see we have been using `warnings` before for this purpose, but we actually have a sophisticated logging config.
The logging system also allows to issue warnings. We can also simply integrate its use with our logging system: https://docs.python.org/3/library/logging.html#logging.captureWarnings

To test, we could capture logging messages. I believe we already do that somewhere.

We can also postpone this decision and leave this PR alone.

Reply:
The blueprint deprecation decorator uses logging instead of warnings. I'll change it to logging for now, but we can discuss which option to choose for the whole project later.